[Docs] Spacing & layout - content restructure (2 of 5) (#5573)
grommet/hpe-design-system: aries-site/src/pages/foundation/spacing.mdx

import { Notification, Paragraph } from 'grommet';
import { LinkNext } from '@hpe-design/icons-grommet';
import { ContentPane, BestPracticeGroup, Example } from '../../layouts';
import { SpacingBestPractices, SpacingScale, SpacingTokensCTA } from '../../examples';

Spacing sizes are a subset of the [scale system](/foundation/scale-system)
using proximity principles to define relationships between content and create
hierarchy within a page. Spacing design tokens define the padding within, margin

> around, or gap between content.

Reviewer: ```suggestion
Spacing sizes are a subset of the [scale system](/foundation/scale-system),
supporting proximity principles to shape relationships between content and establish hierarchy on a page. These principles are applied through [spacing tokens](/design-tokens/layout-and-spacing/spacing), which define the padding within, margin
around, or gap between content.
```

Spacing guides users' eyes through the page, helping them understand the relationships
between different elements. The HPE Design System minimizes use of borders to separate
content. Instead spacing, typography, and background color are combined to create visual

> hierarchy.

Reviewer 2: ```suggestion
Effective spacing guides a user's eye through the interface, making it easier to understand how elements relate to one another. The HPE Design System minimizes use of borders to separate
content. Instead spacing, typography, and background color are combined to create clear visual
hierarchy.
```

## Spacing scale

The following spacing scale may be applied to create separation between [content
containers](/insert-link-when-available) (gap), between a content container's boundary

> and its content (padding), and within content subsections (gap).

Reviewer 2: ```suggestion
Use spacing to:

- Create separation between content containers (gaps)
- Set space between a container’s boundary and its content (padding)
- Define spacing within subsections of content (gaps)
```

<Notification
  alignSelf="start"
  status="warning"
  title="Avoid use of margin"
  message={
    <Paragraph margin="none">
      Use of margin can lead to unpredictable responsive and/or adaptive layouts
      and should be used with caution, if not avoided. All layouts may be
      achieved using gap and padding, resulting in more flexible, predictable,
      and maintainable layouts.
    </Paragraph>
  }

> />

Author: I like this idea. Do you have a supporting graphic / illustration in mind? I am blanking on a illustrative scenario at the moment.

> />

Reviewer: @halocline where did we land on this? do you need me to send you images or is this something you can add on the code side?

<br />
<Example
  caption="Available spacing scale dimensions by t-shirt size"
  height="none"
>
  <SpacingScale />
</Example>

<SpacingTokensCTA />

## Spacing best practices

<BestPracticeGroup>
  <Example
    height="none"
    bestPractice={{
      type: 'do',
      message: `Use spacing design tokens to style compositional or layout components 
        like cards, layers, and grid spacing.`,
    }}
    width="100%"
  >
    <SpacingBestPractices />
  </Example>
  <Example
    height="none"
    bestPractice={{
      type: 'dont',
      message: `Use spacing design tokens to style element-level components. 
      Instead, use component-specific tokens or element tokens which ensure 

>       correct sizing of elements across screen sizes.`,

Reviewer 2: ```suggestion
      message: `Use spacing design tokens to style element-level components. 
      Instead, use component-specific tokens or [element tokens](https://design-system.hpe.design/design-tokens/element) which ensure 
      correct sizing of elements across screen sizes.`,
```

Author: Adopted.
